packages/mobile/src/screens/edit-track-screen/components/CollectibleGatedAvailability.tsx

```diff
@@ -253,18 +258,18 @@ export const CollectibleGatedAvailability = ({
             </Text>
             <IconCaretRight fill={neutralLight4} width={16} height={16} />
           </View>
-          {premiumConditions?.nft_collection && (
+          {isPremiumContentCollectibleGated(premiumConditions) && (
             <View>
               <Text style={styles.ownersOf}>{messages.ownersOf}</Text>
               <View style={styles.collection}>
-                {premiumConditions.nft_collection.imageUrl && (
+                {premiumConditions.nft_collection?.imageUrl && (
```

Comment:
Is the `?` necessary here? I would expect that the type guard function narrows it to a type here where we know that `nft_collection` is defined.

Reply:
Unfortunately @saliou and I discovered that `nft_collection` specifically can be undefined, bc during upload flow user could set the track to collectible gated without specifying a collection yet. Kind of awkward because at every point afterwards all the fields should be defined.

Reply:
will include comment

Reply:
Ouch. That does make this whole change a little less useful.

Reply:
yeah it was a bummer :( couldn't think of a way around it. still gonna be useful for usdc tho.